### PR TITLE
Attempt to fix Posix context switching after lazy init changes

### PR DESCRIPTION
**Pull request by hkaiser**
Blind edit of the POSIX context switching, fixing it after lazy init changes

### CODE REVIEW

File: hpx/runtime/threads/coroutines/detail/context_posix.hpp [hunk 1]
```diff
@@ -396,7 +408,6 @@ namespace hpx { namespace threads { namespace coroutines
             // declare m_stack_size first so we can use it to initialize m_stack
             std::ptrdiff_t m_stack_size;
             void * m_stack;
-            void * cb_;
```
Comment (sithhell) [inline hpx/runtime/threads/coroutines/detail/context_posix.hpp]: why has this been removed?

**Comment by khuck** [comment]
@hkaiser @sithhell I committed the changes to fix this branch.  It builds with Clang 7 on POWER8 now.

**Comment by hkaiser** [comment]
@khuck thanks a lot!

File: hpx/runtime/threads/coroutines/detail/context_posix.hpp (hunk 1 above)
Comment (hkaiser) [reply hpx/runtime/threads/coroutines/detail/context_posix.hpp]: @sithhell do we still need it?

Comment (sithhell) [reply hpx/runtime/threads/coroutines/detail/context_posix.hpp]: oh right, missed that we don't need it anymore...

**Pull request merged**